gun-violence-dashboard/src/pages/Homepage.js

```diff
@@ -1,8 +1,91 @@
-import React from "react";
+import React, { useEffect, useState } from "react";
+import USAMap from "react-usa-map";
+import axios from 'axios';
+import states from '../constants/map_constants'
+
+function Homepage() {
+    const [avgStateGunViolence, setAvgStateGunViolence] = useState([]);
+    const [yearForData, setYearForData] = useState([])
+    let stateAbbreviationConversion = new Map(states);
+    const mapOfAverages = new Map();
+    const gunLawUrl = 'https://giffords.org/lawcenter/gun-laws/states/'
+
+    useEffect(() => {
+        async function fetchAvgData() {
+            const url = `https://datausa.io/api/data/?drilldowns=State&measures=Firearm Fatalities&year=${yearForData}`;
+            return axios.get(url)
+                .then((data) => setAvgStateGunViolence(data.data.data))
+                .catch((error) => console.log(error))
+        }
+
+        fetchAvgData()
+
+    }, [yearForData]);
+
+    const giveStatesTitles = () => {
+        const map = document.querySelector('.us-state-map')
+        const mapTitles = map.querySelectorAll('title');
+        mapTitles.forEach((mapTitle) => {
+            let fatalityValue = mapOfAverages.get(mapTitle.textContent);
+            if (fatalityValue)
+                mapTitle.textContent = `Average Firearm Fatality for ${mapTitle.textContent}: ${String(fatalityValue)}`;
+        })
+    }
+
+    const formatStates = () => {
+        let colorMap = {};
+        if (avgStateGunViolence.length > 0) {
+            avgStateGunViolence.forEach((st) => {
+                const stateName = st['State']
+                const opacity = Number((st['Firearm Fatalities'] / 100)).toFixed(4) * 4;
+                const abbreviation = stateAbbreviationConversion.get(stateName);
+                if (abbreviation === null || abbreviation === '') {
+                    console.log("ERROR", st)
+                }
+                else {
+                    let fill = `rgba(240 , 0, 0, ${opacity} )`
+                    colorMap[abbreviation] = {
+                        fill: fill,
+                        clickHandler: (event) => window.location.href = gunLawUrl + stateName
+                    }
+                }
+                mapOfAverages.set(stateName, st['Firearm Fatalities'].toFixed(2));
```

Comment:
this loop could be broken out into a new function I think, would make it easier to follow